### PR TITLE
feat(vercel): Disable integration completion until a project is linked

### PR DESCRIPTION
**Pull request by leeandher**
This PR disables the Complete on Vercel button until at least one project is linked

**New**

image
image

### CODE REVIEW

**Review by scefali — CHANGES_REQUESTED** [review]
Can you split up the BE and FE for this change since they are coupled?

File: static/app/components/forms/projectMapperField.tsx [hunk 1]
```diff
@@ -279,15 +305,13 @@ export class RenderField extends Component<RenderProps, State> {
...
-                size="small"
-                priority="primary"
-                icon={<IconOpen size="xs" color="white" />}
-                href={nextUrl}
-              >
-                {nextButtonText}
-              </Button>
+              {disabledTooltipText ? (
+                <Tooltip title={disabledTooltipText} disabled={!isNextButtonDisabled}>
+                  {<NextButton href={nextUrl} />}
```
Comment (scefali) [inline static/app/components/forms/projectMapperField.tsx:310]: @leeandher theres no prop for a tooltip title? I don’t think you actually need to wrap the button component like this.

Comment (leeandher) [reply static/app/components/forms/projectMapperField.tsx:310]: Didn't know there was a props for this, thanks!

File: src/sentry/integrations/vercel/integration.py [hunk 2]
```diff
@@ -178,6 +178,8 @@ def get_organization_config(self):
                         "Link your Sentry projects to complete your installation on Vercel"
                     ),
                     "text": _("Complete on Vercel"),
+                    "requiresMapping": True,
```
[the remark this answers is not in the document]
Comment (leeandher) [reply src/sentry/integrations/vercel/integration.py:181]: Sure, it'd make it one FE PR as well. That makes sense

**Review by scefali — APPROVED** [review]

**Pull request merged**